journal.py

```diff
@@ -102,69 +155,90 @@ def update_entry(request, id):
-    return HTTPFound(request.route_url('home'))
-
+    if request.authenticated_userid:
+        if request.method == 'POST':
+            try:
+                Entry.from_request(request)
+            # This error type needs to be updated since it will now be an SQLA error
+            # type.
+            except psycopg2.Error:
+                # this will catch any errors generated by the database
+                return HTTPInternalServerError
+            return get_newest_entry(request)
+    else:
+        return HTTPForbidden()
 
 # HOME PAGE
 @view_config(route_name='home', renderer='templates/list.jinja2')
 def read_entries(request):
     """return a list of all entries as dicts"""
-    cursor = request.db.cursor()
-    cursor.execute(DB_ENTRIES_LIST)
-    keys = ('id', 'title', 'text', 'created')
-    entries = [dict(zip(keys, row)) for row in cursor.fetchall()]
+    entries = Entry.all()
     for entry in entries:
-        entry['text'] = mdown(entry['text'])
+        entry.text = mdown(entry.text)
     return {'entries': entries}
 
 
 # EDIT ENTRY PAGE
-@view_config(route_name='edit', renderer='templates/edit.jinja2')
+@view_config(route_name='edit', renderer='json')
 def edit(request):
     """view for edit single entry"""
+    entry = ""
+    id = request.params.get('id', None)
     if request.authenticated_userid:
         if request.method == 'POST':
             try:
-                id = request.matchdict.get('id', -1)
                 update_entry(request, id)
+                entry = get_single_entry(request, False, True, id)
             except psycopg2.Error:
                 # this will catch any errors generated by the database
                 return HTTPInternalServerError
-            return HTTPFound(request.route_url('home'))
-    return get_single_entry(request)
+        else: # GET REQUEST
+            # import pdb; pdb.set_trace()
+            entry = get_single_entry(request, False, True, id)
+    entry = to_json(entry)
+    entry['created'] = entry['created'].strftime('%b %d, %Y')
+    return entry
 
 
 # DETAIL PAGE
 @view_config(route_name='detail', renderer='templates/detail.jinja2')
 def detail_view(request):
     """view for entry (permalink for an entry)."""
-    return get_single_entry(request, True)
-
-
-def get_single_entry(request, mark_down=False):
-    """get single entry - returns markdown if mark_down set to True"""
     id = request.matchdict.get('id', -1)
-    cursor = request.db.cursor()
-    cursor.execute(SELECT_SINGLE_ENTRY, (id,))
-    keys = ('id', 'title', 'text', 'created')
-    entry = dict(zip(keys, cursor.fetchone()))
+    return get_single_entry(request, True, False, id)
+
+def get_newest_entry(request):
+    entry = Entry.get_newest()
+    # cursor = request.db.cursor()
+    # cursor.execute(SELECT_SINGLE_NEWEST_ENTRY)
+    # keys = ('id', 'title', 'text', 'created')
+    # entry = dict(zip(keys, cursor.fetchone()))
+    entry.text = mdown(entry.text)
```

Comment:
Here again you are altering the entry object's `text` in-place, and if the request/response cycle completes without error, that change will be persisted to the database.